processing/src/main/java/io/druid/query/ordering/StringComparators.java

```diff
@@ -342,16 +329,61 @@ public String toString()
     }
   }
 
-  public static StringComparator makeComparator(String type)
+  private static BigDecimal convertStringToBigDecimal(String input) {
+    // treat unparseable Strings as nulls
+    BigDecimal bd = null;
+    try {
+      bd = new BigDecimal(input);
+    } catch (NullPointerException | NumberFormatException ex) {
+    }
+    return bd;
+  }
+
+  public static class NumericComparator extends StringComparator
   {
-    if (type.equals(StringComparators.LEXICOGRAPHIC_NAME)) {
-      return LEXICOGRAPHIC;
-    } else if (type.equals(StringComparators.ALPHANUMERIC_NAME)) {
-      return ALPHANUMERIC;
-    } else if (type.equals(StringComparators.STRLEN_NAME)) {
-      return STRLEN;
-    } else {
-      throw new IAE("Unknown string comparator[%s]", type);
+    @Override
+    public int compare(String o1, String o2)
```

Comment:
@gianm I've added two new queries to the TopNBenchmark that use numeric/alphanumeric sort.

Benchmark results have been posted to the main comment.